src/probnum/linops/_kronecker.py

```diff
@@ -161,6 +161,44 @@ def _cond(self, p) -> np.inexact:
 
         return np.linalg.cond(self.todense(cache=False), p=p)
 
+    def _matmul_kronecker(self, other: "Kronecker") -> "Kronecker":
+        if self.shape[1] != other.shape[0]:
+            raise ValueError(
+                f"matmul received invalid shapes {self.shape} @ {other.shape}"
+            )
+
+        _A, _B = self.A, self.B
+        _C, _D = other.A, other.B
+        if not (_A.shape[1] == _C.shape[0] and _B.shape[1] == _D.shape[0]):
+            return NotImplemented
+
+        # Using (A (x) B) @ (C (x) D) = (A @ C) (x) (B @ D)
+        return Kronecker(A=_A @ _C, B=_B @ _D)
+
+    def _add_kronecker(
+        self, other: "Kronecker"
+    ) -> Union[NotImplementedType, "Kronecker"]:
+
+        if self.A == other.A:
```

Comment:
An additional check for `self.A is other.A` covers even more cases, even without the `__eq__` operator implemented